types/lib/amazon-pay.d.ts

```diff
@@ -0,0 +1,46 @@
+import { Emitter } from './emitter';
+
+export type AmazonPayOptions = {
+  /**
+   * 2 Digit Country Code
+   */
+  region: string;
+
+  /**
+   * Specify which Payment Gateway in Recurly must handle the payment.
+   */
+  gatewayCode?: string
+
+  /**
+   * Sets button to Sandbox environment
+   */
+  sandbox?: boolean;
+};
+
+export type AmazonPayRenderButtonOptions = {
+  productType?: string;
+  placement?: string;
+  buttonColor?: string;
+};
+
+export type AmazonPayEvent = 'ready' | 'token' | 'error' | 'close' | 'done';
+
+export interface AmazonPayInstance extends Emitter<AmazonPayEvent> {
+  /**
+   * Invokes the Amazon Payment Modal
+   */
+  start: (amazonPayOptions: AmazonPayOptions) => void;
+
+  /**
+   * Attaches an Element to the DOM, as a child of the specified parent target.
+   *
+   */
+  attach: (defaultEventName: string, gatewayCode?: string) => AmazonPayInstance;
```

Comment:
The `attach` function does not accept any parameters and does not return anything.